src/shared/containers/Refile/RefileErrorsForm.jsx

```diff
@@ -0,0 +1,236 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import axios from  'axios';
+import isEmpty from 'lodash/isEmpty';
+import forIn from 'lodash/forIn';
+import FormField from '../../components/FormField/FormField';
+
+class RefileErrorsForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      formFields: {
+        startDate: '',
+        endDate: '',
+      },
+      fieldErrors: {},
+      formResult: {}
+    };
+    this.baseState = this.state;
+    this.handleInputChange = this.handleInputChange.bind(this);
+    this.handleInputBlur = this.handleInputBlur.bind(this);
+    this.handleFormSubmit = this.handleFormSubmit.bind(this);
+  }
+
+  /**
+  * @desc Handles updating the state for the given field name based on the value changes
+  * @param {object} event - contains the current event context of the input field
+  */
+  handleInputChange(event) {
+    const target = event.target;
+    const value = target.value;
+    const name = target.name;
+    this.setState({ formFields: {...this.state.formFields, [name]: value} });
+  }
+
+  /**
+  * @desc Handles updating the state based on the validation function executed in the setState
+  * anonymous function.
+  * @param {object} event - contains the current event context of the input field
+  */
+  handleInputBlur(event) {
```

Comment:
Not suggesting you change this now, but feels like there is a lot of boilerplate here — the `setState`, `focusOnField` methods, etc., are repeated. Worth introducing a mixin or superclass at some point?

Reply:
`setState` is the function comes from `react` to update the state, so probably can't really make a mixin out of it.
https://reactjs.org/docs/react-component.html#setstate

About other functions, I am still on the way to simplify the components. You can see them in the later PRs.

Reply:
Awesome. Sounds great!